packages/usa-combo-box/src/index.js

```diff
@@ -382,28 +382,63 @@ const displayList = (el) => {
   const filter = comboBoxEl.dataset.filter || DEFAULT_FILTER;
   const regex = generateDynamicRegExp(filter, inputValue, comboBoxEl.dataset);
 
-  const options = [];
-  for (let i = 0, len = selectEl.options.length; i < len; i += 1) {
-    const optionEl = selectEl.options[i];
-    const optionId = `${listOptionBaseId}${options.length}`;
-
-    if (
-      optionEl.value &&
-      (disableFiltering ||
-        isPristine ||
-        !inputValue ||
-        regex.test(optionEl.text))
-    ) {
-      if (selectEl.value && optionEl.value === selectEl.value) {
-        selectedItemId = optionId;
+  let options = [];
+  const optionsAlphabetical = [];
+  const optionsContains = [];
+  const optionList = [...selectEl.options];
+
+  /**
+   * Builds options array.
+   *
+   * If disableFiltering or a selection has been made: do not filter.
+   * Else filter options and sort by alphabetical matches followed by
+   * options that contain the query.
+   *
+   * @param {HTMLOptionElement} option - Option element from select array
+   */
+  const buildOptionsArray = (option) => {
+    if (disableFiltering || isPristine) {
+      options.push(option);
```

Comment:
**note:** adding `isPristine` here prevents selected values from appearing at the top of the combo box. This allows the selected item to appear in the correct order, which matches current combo box behavior